src/hydra-queue-runner/hydra-queue-runner.cc

```diff
@@ -742,6 +750,24 @@ void State::unlock()
 }
 
 
+void State::runMetricsExporter()
+{
+    std::cout << "Starting the Prometheus exporter on port " << metricsPort << std::endl;
+
+    /* Set up simple exporter, to show that we're still alive. */
+    std::string metricsAddress{"127.0.0.1"};
+    prometheus::Exposer exposer{metricsAddress + ":" + std::to_string(metricsPort)};
+    auto exposerPort = exposer.GetListeningPorts().front();
+    exposer.RegisterCollectable(registry);
+
+    std::cout << "Started the Prometheus exporter, listening on "
+        << "http://" << metricsAddress << ":" << exposerPort << "/metrics"
+        << std::endl;
+
+    while (true) {};
```

Comment:
This needs to run forever, otherwise the `prometheus::Exposer` gets dropped at the end of this scope.